systemtest/src/test/java/io/strimzi/systemtest/performance/TopicOperatorPerformanceTest.java

```diff
@@ -0,0 +1,611 @@
+            this.daemonThread = new Thread(this.topicOperatorPollingThread);
+            this.daemonThread.setDaemon(true); // Set as daemon so it doesn't prevent JVM shutdown
+            this.daemonThread.start();
+            // ----- ----- ------ ------
+
+            // Measure topic creation time
+            startTimeMs = System.currentTimeMillis();
+
+            // Create topics
+            KafkaTopicScalabilityUtils.createTopicsViaK8s(testStorage.getNamespaceName(), testStorage.getClusterName(), topicNamePrefix,
+                    numberOfTopics, 12, 3, 1);
+
+            KafkaTopicScalabilityUtils.waitForTopicsReady(testStorage.getNamespaceName(), topicNamePrefix, numberOfTopics);
+
+            endTimeMs = System.currentTimeMillis();
+            createTopicsTimeMs = endTimeMs - startTimeMs;
+            LOGGER.info("Time taken to create {} topics: {} ms", numberOfTopics, createTopicsTimeMs);
+
+            endTimeWholeMs = System.currentTimeMillis();
+            totalTimeWholeMs = endTimeWholeMs - startTimeMs;
+
+            LOGGER.info("Time taken to create {} topics and send and recv: {} messages: {} ms", numberOfTopics, NUMBER_OF_MESSAGES, totalTimeWholeMs);
+
+            if (withClientsEnabled) {
+                // delete clients here
+                for (int i = 0; i < numberOfClientInstances; i++) {
+                    ClientUtils.waitForClientSuccess(testStorage.getProducerName() + "-" + i, testStorage.getNamespaceName(), NUMBER_OF_MESSAGES);
+                    ClientUtils.waitForClientSuccess(testStorage.getConsumerName() + "-" + i, testStorage.getNamespaceName(), NUMBER_OF_MESSAGES);
+                }
+                endSendRecvTimeMs = System.currentTimeMillis();
+                totalSendAndRecvTimeMs = endSendRecvTimeMs - startSendRecvTimeMs;
+
+                LOGGER.info("Time taken send and recv: {} messages: {} ms", NUMBER_OF_MESSAGES, totalSendAndRecvTimeMs);
+            }
+
+            // Start measuring time for deletion of all topics
+            long deletionStartTimeMs = System.currentTimeMillis();
+
+            LOGGER.info("Start deletion of {} KafkaTopics in namespace:{}", numberOfTopics, testStorage.getNamespaceName());
+            resourceManager.deleteResourcesOfTypeWithoutWait(KafkaTopic.RESOURCE_KIND);
+            KafkaTopicUtils.waitForTopicWithPrefixDeletion(testStorage.getNamespaceName(), topicNamePrefix);
+
+            long deletionEndTime = System.currentTimeMillis();
+            totalDeletionTimeMs = deletionEndTime - deletionStartTimeMs;
+            LOGGER.info("Time taken to delete {} topics: {} ms", numberOfTopics, totalDeletionTimeMs);
+
+            endTimeWholeMs = System.currentTimeMillis();
+            totalTimeWholeMs = endTimeWholeMs - startTimeMs;
+
+            LOGGER.info("Total time taken to create {} topics, send and receive {} messages, and delete topics: {} ms", numberOfTopics, NUMBER_OF_MESSAGES, totalTimeWholeMs);
+        } finally {
```

Comment:
why is it whole in the try catch block?

Reply:
If you take a closer look..that's not a `try catch` but only `try with finally` where finally ensures that `topicOperatorThread` is always finished gracefully. same for the second one.

Reply:
I'm just thinking -> do you think that there is a possibility to have this graceful termination inside the `topicOperatorThread` and either call it on after each or in case of some interruption? I think it would be good to have it that way, so you don't need to handle it in each test case - but that's just an idea :)

Reply:
Now as I re-wrote it into `ScheduledExecutorService` I have to call it always in finally before:
```java
final Map<String, Object> performanceAttributes = new LinkedHashMap<>();
                performanceAttributes.put(PerformanceConstants.NUMBER_OF_TOPICS, numberOfTopics);
                performanceAttributes.put(PerformanceConstants.NUMBER_OF_CLIENT_INSTANCES, numberOfClientInstances);
                performanceAttributes.put(PerformanceConstants.NUMBER_OF_MESSAGES, NUMBER_OF_MESSAGES);
                performanceAttributes.put(PerformanceConstants.CREATION_TIME, createTopicsTimeMs + " ms");
                performanceAttributes.put(PerformanceConstants.SEND_AND_RECV_TIME, totalSendAndRecvTimeMs + " ms");
                performanceAttributes.put(PerformanceConstants.DELETION_TIME, totalDeletionTimeMs + " ms");
                performanceAttributes.put(PerformanceConstants.TOTAL_TEST_TIME, totalTimeWholeMs + " ms");
                performanceAttributes.put(PerformanceConstants.MAX_BATCH_SIZE, maxBatchSize);
                performanceAttributes.put(PerformanceConstants.MAX_BATCH_LINGER_MS, maxBatchLingerMs);

                // Handling complex objects
                performanceAttributes.put(PerformanceConstants.METRICS_HISTORY, this.topicOperatorMetricsGatherer.getMetricsHistory()); // Map of metrics history

                // Step 3: Now, it's safe to log performance data as the collection thread has been stopped
                this.topicOperatorPerformanceReporter.logPerformanceData(this.testStorage, performanceAttributes, TopicOperatorPerformanceTest.REPORT_DIRECTORY + "/" + PerformanceConstants.TOPIC_OPERATOR_ALICE_BULK_USE_CASE, ACTUAL_TIME, Environment.PERFORMANCE_DIR);
```
because if I don't do that it may happen that there will be `ConcurrentExpectionModification` in metrics files where I would be iterating over it and collector would change that... :/ 

So I think it would be perfect for each test case start and stop collecting (i.e., in before and after each) but 
```java
// -- Metrics POLL --
            // Assuming 'testStorage' contains necessary details like namespace and scraperPodName
            this.topicOperatorCollector = new TopicOperatorMetricsCollector.Builder()
                    .withScraperPodName(this.testStorage.getScraperPodName())
                    .withNamespaceName(this.testStorage.getNamespaceName())
                    .withComponentType(ComponentType.TopicOperator)
                    .withComponentName(this.testStorage.getClusterName())
                    .build();

            this.topicOperatorMetricsGatherer = new TopicOperatorMetricsGatherer(this.topicOperatorCollector, "strimzi.io/cluster=" + this.testStorage.getClusterName());
            this.topicOperatorMetricsGatherer.startCollecting();
```
Would fail if TopicOperator is not up and running...:
1. Okay we can use wait until....if we deploy this at the start...but how much should we wait? That's kinda hard because each infra is different...
...
the second problem would be `AfterEach` and that race condition:
1. We could make `topicOperatorPerformanceReporter` instance thread-safe (synchronize) -> that would possibly work...
---

Reply:
I am not sure we want such a change in this PR :) (i.e., making transparent a collector)

Reply:
I'm fine with leaving it for future PR, thanks :)